src/Processors/Formats/Impl/ParquetBlockOutputFormat.cpp

```diff
@@ -282,11 +281,17 @@ void ParquetBlockOutputFormat::writeRowGroup(std::vector<Chunk> chunks)
         writeUsingArrow(std::move(chunks));
     else
     {
-        Chunk concatenated = std::move(chunks[0]);
-        for (size_t i = 1; i < chunks.size(); ++i)
-            concatenated.append(chunks[i]);
+        Chunk concatenated;
+        while (!chunks.empty())
+        {
+            chunks_count++;
+            if (concatenated.empty())
+                concatenated = std::move(chunks.back());
+            else
+                concatenated.append(chunks.back());
```

Comment:
Why no `std::move` here?


```suggestion
                concatenated.append(std::move(chunks.back()));
```

Reply:
Because `void append(const Chunk & chunk);` accepts const reference and move doesn't makes sense. Maybe clearing `chunks.back()` still makes sense, so we won't keep it in memory after it's not needed anymore and release a bit more quickly. But not sure if ti's worth it, current version should be good enough